overlord/overlord.go

```diff
@@ -93,6 +101,34 @@ func loadState(backend state.Backend) (*state.State, error) {
 	return state.ReadState(backend, r)
 }
 
+// Run runs a loop to ensure the current state regularly through StateEngine Ensure().
+func (o *Overlord) Run() {
+	intv := ensureInterval
+	o.loopTomb.Go(func() error {
+		timer := time.NewTimer(intv)
+		for {
+			select {
+			case <-o.loopTomb.Dying():
+				return nil
+			case <-timer.C:
+			}
+			err := o.stateEng.Ensure()
+			if err != nil {
+				logger.Noticef("state engine ensure failed: %v", err)
+				// continue to the next Ensure() try for now
+			}
+			timer.Reset(intv)
```

Comment:
Not sure if it makes sense to reset it here. It means we're delaying the execution of the first managers based how much everybody took to execute. The intended semantics we want in the end is likely "every N minutes", no matter how much time it takes to execute. If removing the Reset means we have a loop that is too tight, we can increase the initial delay a bit (5 seconds?).